repo-client/src/main/java/io/lionweb/repoclient/api/BulkAPIClient.java

```diff
@@ -3,17 +3,21 @@
 import io.lionweb.lioncore.java.model.Node;
 import java.io.IOException;
 import java.util.List;
+import org.jetbrains.annotations.Nullable;
 
 public interface BulkAPIClient {
-  void createPartitions(List<Node> partitions) throws IOException;
+  @Nullable
+  RepositoryVersionToken createPartitions(List<Node> partitions) throws IOException;
```

Comment:
If we separate regular client from history client, do we want to introduce history stuff (`RepositoryVersionToken`) to the regular client?

Reply:
We have separate interfaces: one for each module of the repository.

We do that for two reasons:

* because of a previous comment by Meinte on having a more modular implementation
* because in this way, one could introduce _other implementations_ for a subset of the interface

For the implementation currently in this codebase I think it makes sense to support all the interfaces because most methods work both for repository with or without history.

Reply:
I was thinking whether we want to have anything history-related in the non-history Interface. We still can implement all interfaces on one class.

Reply:
The non-history interface is extended by the history one. We could duplicate the methods such as listPartitions and createPartitions, but then one would need to call different methods depending on the fact they are operating on a repo with history or without. It seems to be over complicated. I thought about that for one week but I could not find a proper solution, so I would lean toward merging this as-is